ext/cl_khr_defined_builtin_kernels.asciidoc

```diff
@@ -99,222 +103,414 @@ accelerators while being able to synchronize and optimize data transfers at
+
+* *Name*: Name of the defined built-in kernel (an enumeration).
+
+* *Kernel attributes*: The kernel attributes required for creating the
+  defined built-in kernel via
+  clCreateDefinedBuiltInKernelDescriptor. Attribute values are
+  immutable.
+
+* *Kernel arguments*: The kernel arguments.
+
+* *Description*: The description of the kernel in detail.
+
+* *Attribute validation rules*: Conditions of the kernel attribute for
+  the kernel. Implementation must return CL_DBK_INVALID_ATTRIBUTE on
+  clCreateDefinedBuiltInKernelDescriptor call if any of the conditions
+  are violated.
+
+* *Kernel mode properties*: List of kernel mode
+   properties (cl_dbk_mode_properties) the kernel recognizes. The
+   properties can be used to tweak certain implementation details and
+   behaviors in the kernel execution. If a property not listed in the
+   DBK entry is fed to clCreateDefinedBuiltInKernelDescriptor call,
+   then implementation must return CL_DKB_UNSUPPORTED_MODE_PROPERTY.
 
 [caption="Table A.I.1. "]
 .Standard Built-in Kernels and Their Semantics. *The table has been populated with a small set of non-trivial example entries which are subject to change and the list to expand during drafting.*
-[cols="1,3,2,2"]
 |===
-4+| *General linear algebra*
-// https://netlib.org/blas/blasqr.pdf
-| Name | Description | NDRange Dimensions | Arguments
-| *khr_blas_gemm_float*
-| xGEMM: General matrix multiplication with real single precision floating point numbers as described in Basic Linear Algebra Subprograms. Performs C = alpha * trans(A) * trans(B) + beta*C, where A, B and C are matrices, and alpha and beta scalars. trans() is a configurable transpose operation.
-a|
-[start=1]
-. The height.
-. The width.
+| Name: *khr_matmul_v1*
+| *Kernel Attributes*
 a|
-[start=0]
-. int: transpose operation (trans) type for matrix A (0 = none, 1 = transpose, 2 = conjugate transpose)
-. int: transpose type for matrix B (0 = none, 1 = transpose, 2 = conjugate transpose)
-. float: scalar (alpha) to multiply the matrix multiplication result elements with
-. float* (input): matrix A
-. int: leading dimension of A (0 = row-major, 1 = column-major)
-. float* (input): matrix B
-. int: leading dimension of B (0 = row-major, 1 = column-major)
-. float: scalar (beta) to multiply the C matrix elements with before adding it to the result
-. float* (input&output): matrix C which is added to the matrix multiplication result, and stores the output
-. int: leading dimension of C (0 = row-major, 1 = column-major)
-4+| OpenCL C Semantics
-4+a|
-[source,c]
-----
-__kernel void __khr_blas_gemm_float(
-   int transA, int transB, float alpha, const global float *A, int ldA,
-   const global float *B, int ldB,
-   float beta, global float *C, int ldC) {
-   // TBD: An example implementation that can be used for verification
-   // and as a fallback SW implementation.
-}
-----
 
-4+| *OpenVX Neural Network Extension Compatible Kernels*
```

Comment:
This is a good reminder. We should ensure we can implement OpenVX NN extension nicely with the DBKs to keep compatibility with the other Khronos APIs.

Reply:
(And also other common compuer vision algorithms.)